padl/dumptools/symfinder.py

```diff
@@ -607,39 +608,50 @@ def f(x):
     scope: Scope
     n: int = 0
```

Comment:
Also please add some explanation about the `variants` in the docstring.

Reply:
Right now I have kept dataclass for the convenience of the printing. I don't see any harm keeping it or big advantage removing it.

Reply:
Yeah, the point was that now that the class has so much extra stuff there's also not that much of a benefit any more, and in my view if there's not much of an advantage it's better to just have an ordinary class (because people are more familiar with it).